### ISSUE TITLE
Cannot login to Microsoft account for synchronization

### ISSUE THREAD

**Issue by githubyaner** [microsoft/vscode#180812]
Type: <b>Bug</b>

I reinstalled Vscode, clicked on "Turn On Setting Sync", selected "Sign in & Turn on",   then selected " Sign in with Microsoft".
When I finish logging in at the browser, the screen says "You are signed in now and can close this page.".
Vscode is unresponsive. "Error while turning on Settings Sync. Network failure
" was shown.

VS Code version: Code 1.77.3 (Universal) (704ed70d4fd1c6bd6342c436f1ede30d1cff4710, 2023-04-12T09:19:37.325Z)
OS version: Darwin x64 22.4.0
Modes:
Sandboxed: Yes

**Comment by githubyaner** [comment]
Thank you for your guidance, I followed your steps and found the log content.
```
2023-04-26 00:04:17.457 [info] Getting sessions for the following scopes: email offline_access openid profile
2023-04-26 00:04:17.457 [trace] No session found with idtoken scopes... Using fallback scope list of: email offline_access openid profile
2023-04-26 00:04:17.458 [info] Got 0 sessions for scopes: email offline_access openid profile
2023-04-26 00:04:18.839 [info] Logging in for the following scopes: email offline_access openid profile
2023-04-26 00:04:24.530 [info] Exchanging login code for token for scopes: email offline_access openid profile
2023-04-26 00:05:36.014 [error] Fetching token failed for scopes (email offline_access openid profile): request to https://login.microsoftonline.com/organizations/oauth2/v2.0/token failed, reason: certificate has expired
2023-04-26 00:05:36.015 [error] Error exchanging code for token (for scopes email offline_access openid profile): Error: Network failure

```

**Comment by TylerLeonhardt** [comment]
@githubyaner on this machine are you behind a Corporate Proxy/Firewall?

**Comment by githubyaner** [comment]
Yes.But I use my personal network with this device and still have this problem.
Using the github login is fine.

**Comment by TylerLeonhardt** [comment]
Can you try out @chrmarti's Proxy debugging extension here: https://github.com/microsoft/vscode-remote-release/issues/8248#issuecomment-1494127068

and let me know how it goes?

**Comment by githubyaner** [comment]
@TylerLeonhardt 
https://github.com/microsoft/vscode-remote-release/issues/8248#issuecomment-1494127068
I followed the steps in the 👆🏻 link, and it didn't seem to work.
My accidental action caused the keychain exception, but I'm not sure if it was the keychain exception that caused it.
I also ignored my network and re-linked it.
Finally, I deleted vscode completely, restarted the computer, and reinstalled it. I was able to synchronize the Microsoft account login.
The problem is solved. Thank you very much for your patience and guidance.
